automation/entry_scripts/balena-build-block.sh

```diff
@@ -0,0 +1,115 @@
+#!/bin/bash
+
+set -e
+
+script_dir=$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )
+source "${script_dir}/balena-api.inc"
+source "${script_dir}/balena-lib.inc"
+
+# Input checks
+[ -z "${APPNAME}" ] && echo "The block's app  name needs to be defined" && exit 1
+[ -z "${API_ENV}" ] && echo "Balena's environment needs to be defined" && exit 1
+[ -z "${BALENAOS_TOKEN}" ] && echo "Balena's environment token needs to be defined" && exit 1
+[ -z "${MACHINE}" ] && echo "Machine needs to be defined" && exit 1
+[ -z "${PACKAGES}" ] && echo "list of packages to install without dependencies" && exit 1
+[ -z "${RELEASE_VERSION}" ] && echo "A release version needs to be defined" && exit 1
+[ -z "${WORKSPACE}" ] && echo "Workspace needs to be defined" && exit 1
+
+[ -z "${PACKAGE_TYPE}" ] && PACKAGE_TYPE="ipk"
+[ -z "${BALENAOS_ACCOUNT}" ] && BALENAOS_ACCOUNT="balena_os"
+
+DEVICE_TYPE_JSON="$WORKSPACE/$MACHINE.json"
+if [ -e "${DEVICE_TYPE_JSON}" ]; then
+	ARCH=$(jq --raw-output '.arch' "$DEVICE_TYPE_JSON")
+fi
+[ -z "${ARCH}" ] && echo "Device architecture is required" && exit 1
+
+source /balena-docker.inc
+
+finish() {
+	balena_docker_stop
+	# Dockerd leaves a mount here
+	if ! umount "${DOCKER_ROOT}"; then
+		umount -l "${DOCKER_ROOT}" || true
+	fi
+	rm -rf "${TMPDIR}"
+}
+trap finish EXIT ERR
+
+# Start docker
+# Data root needs to be on a non-aufs directory to use balena build
+TMPDIR=$(mktemp -d --tmpdir="${WORKSPACE:?}")
+DOCKER_ROOT="${TMPDIR}/docker"
+balena_docker_start "${DOCKER_ROOT}" "/var/run" "/var/log/docker.log"
+balena_docker_wait
+
+# Only support overlay images for the time being. Labels to be parametrized from contract in future.
+cat << 'EOF' > ${TMPDIR}/Dockerfile
+ARG NAMESPACE=resin
+ARG TAG=latest
+FROM ${NAMESPACE}/yocto-block-build-env:${TAG} AS builder
+ARG PACKAGES="space separated list of packages to force install without dependencies"
```

Comment:
What if we didn't set defaults for these args so we can
1. get warnings at build time that args are not set
2. skip the for loop altogether so we don't write nonsense files into opkg and generate weird errors
?

Reply:
Sounds good, I'll change it.